### ISSUE TITLE
Implement possible sensor detection algorithm on XM510

### ISSUE THREAD

**Issue by widgetii** [OpenIPC/ipc_chip_info#5]
Strace output while system starts:

```
/var/utils # ./strace -f -e trace=ioctl dvrHelper /lib/modules /usr/bin/Sofia 127.0.0.1 9578 1 2>&1 |grep ioctl\(9
[pid  1039] ioctl(9, CMD_I2C_WRITE, {dev_addr=0x30, reg_addr=0x12, addr_byte_num=0x1, data=0x80, data_byte_num=0x1}) = 0
[pid  1039] ioctl(9, CMD_I2C_WRITE, {dev_addr=0x30, reg_addr=0x12, addr_byte_num=0x1, data=0x80, data_byte_num=0x1}) = 0
[pid  1039] ioctl(9, CMD_I2C_READ, {dev_addr=0x30, reg_addr=0xa, addr_byte_num=0x1, data=0x3d264, data_byte_num=0x1}) = 0
[pid  1039] ioctl(9, CMD_I2C_READ, {dev_addr=0x30, reg_addr=0xb, addr_byte_num=0x1, data=0xffffffff, data_byte_num=0x1}) = 0
[pid  1039] ioctl(9, CMD_I2C_READ, {dev_addr=0x30, reg_addr=0x3008, addr_byte_num=0x2, data=0xffffffff, data_byte_num=0x1}) = 0
[pid  1039] ioctl(9, CMD_I2C_READ, {dev_addr=0x30, reg_addr=0x3005, addr_byte_num=0x2, data=0xffffffff, data_byte_num=0x1}) = 0
[pid  1039] ioctl(9, CMD_I2C_WRITE, {dev_addr=0x36, reg_addr=0x103, addr_byte_num=0x2, data=0x1, data_byte_num=0x1}) = 0
[pid  1039] ioctl(9, CMD_I2C_READ, {dev_addr=0x36, reg_addr=0x300a, addr_byte_num=0x2, data=0x3d264, data_byte_num=0x1}) = 0
[pid  1039] ioctl(9, CMD_I2C_READ, {dev_addr=0x36, reg_addr=0x300b, addr_byte_num=0x2, data=0xffffffff, data_byte_num=0x1}) = 0
[pid  1039] ioctl(9, CMD_I2C_WRITE, {dev_addr=0x10, reg_addr=0x301a, addr_byte_num=0x2, data=0x1, data_byte_num=0x2}) = 0
[pid  1039] ioctl(9, CMD_I2C_READ, {dev_addr=0x10, reg_addr=0x3000, addr_byte_num=0x2, data=0x3d264, data_byte_num=0x2}) = 0
```

**Comment by widgetii** [comment]
In my case (XM510+AR0130) I got `0x962` for `dev_addr=0x10, reg_addr=0x3000` (last line) and believe this is `AR0130` sensor.

For lines 3-4 it's probably detection algorithm for `Silicon Optronics`

Are lines 5-6 for `Sony`? What's about lines 7-9 with `dev_addr=0x36`?

**Comment by widgetii** [comment]
Fixed in 7cca63a05df8d5db6187892ca58242bea84eab0d
